src/gpu/kmeans/KmMatrix/GpuInfo.cuh

```diff
@@ -0,0 +1,76 @@
+/*!
```

Comment:
This GPU info class is not specific to kmeans. Check how the other algorithms are getting this information then consider moving into a common location and making the other algorithms also use this class. Ideally we want things done consistently across the code base. This makes it much easier to maintain.